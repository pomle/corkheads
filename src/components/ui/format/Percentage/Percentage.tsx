import React from "react";

interface PercentageProps {
  fraction: number;
}

const Percentage: React.FC<PercentageProps> = ({ fraction }) => {
  if (isFinite(fraction)) {
    return <>{(fraction * 100).toFixed()}%</>;
  }
  return <>-</>;
};

export default Percentage;
