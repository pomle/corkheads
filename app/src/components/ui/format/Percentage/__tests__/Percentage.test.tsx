import React from "react";
/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
import renderer from "react-test-renderer";
import Percentage from "../Percentage";

describe("Percentage component", () => {
  it("shows 1 as 100%", () => {
    const component = renderer.create(<Percentage fraction={1} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("rounds number", () => {
    const component = renderer.create(<Percentage fraction={0.54124} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("handles negatives", () => {
    const component = renderer.create(<Percentage fraction={-1.24} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("handles non-finites gracefully", () => {
    const component = renderer.create(<Percentage fraction={1 / 0} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
