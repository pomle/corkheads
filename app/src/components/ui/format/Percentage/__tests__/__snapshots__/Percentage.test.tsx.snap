// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Percentage component handles negatives 1`] = `
Array [
  "-124",
  "%",
]
`;

exports[`Percentage component handles non-finites gracefully 1`] = `"-"`;

exports[`Percentage component rounds number 1`] = `
Array [
  "54",
  "%",
]
`;

exports[`Percentage component shows 1 as 100% 1`] = `
Array [
  "100",
  "%",
]
`;
